### DISCUSSION TITLE
Caribou not working on datas

### DISCUSSION THREAD

**Discussion by simgrouik** [philipbelesky/Caribou#16] [Q&A] [answered]
Hello, I just downloaded Caribou, insert nodes as you example, load an osm file from your data test, but nothing works, no data is out from components.
File had been place inside a "file path" component, and I tried as a single panel element.
I checked and downloaded osm file is well "unblock"
I seems that Caribou is not "processing or reading the file" is is only acquiring the path of it...

Latest Rhino 7.1
![Caribou OSM](https://user-images.githubusercontent.com/46321881/137292682-729f6058-30f8-4c92-bcf3-3f3e5a52a264.PNG)
2

**Comment by philipbelesky** [comment]
Hey, thanks for getting in touch and sorry to hear things aren't working out. Could you wire up some panels to the outputs of the `Ways` component and post a screenshot of what they show (if anything).

If you could also upload the OSM file to this thread I can take a look and see if something is wrong with parsing that particular location.

**Reply by philipbelesky** [reply]
Oh, and one further thing, if your `D:\` drive ins't where Windows is installed, maybe try moving the OSM file to your Documents folder. IIRC there are sometimes file access/security issues loading across drives.

**Comment by simgrouik** [comment]
Hello Philip,

Thank you for your quick answer.
I tried many thing for unfortunately, not yet with an happy issue :(

I use an .osm file from your Github, Copenhagen, place it inside my user document on my C: "windows" drive, check that the file is unblock.

But when I connect, something is made inside Rhino, look at the command panel, and it calculate every time Caribou read a new file, but nothing is out from Caribou node, (I tried with evry options checked, or not, or partially)

Hope it could helps you...

![Caribou OSM2](https://user-images.githubusercontent.com/46321881/137379526-17df978a-47ef-4f45-b0b7-c870b6e955e8.png)

**Reply by philipbelesky** [reply]
Thanks for getting back to me and providing the error printout. I think I've reproduced what at least one of the issues here might be - somewhat bizarrely the file location doesn't read if provided from a panel, but seems to work fine from a `FilePath` component. Have you tried plugging in a `FilePath` component directly? I'll definitely look into why a panel doesn't work and can roll out a fix over the  next few days.

<img width="1016" alt="Screen Shot 2021-10-15 at 11 05 26 am" src="https://user-images.githubusercontent.com/495961/137411642-eb193685-9ae0-46b5-8aa5-2856e67a2bac.png">

**Reply by simgrouik** [reply]
Hello Philip,
Nothing changes... even with a filepath component, I also tried to internalise data... but nothing change, even if I use the Panel as you post in your message I have not the red element, , it works but nothing is out from components...
My idea is that Caribou , maybe use an other driver/library or plugin to work an that this one is not update or in my grasshopper directory...

**Reply by philipbelesky** [reply]
Thanks for more information. I think I've fixed the path issue at least, which should be in a new release out in a few days. 

Regarding your suggestion about the library issue, that is possible. Could I ask how you installed Caribou? Was it via the package manager, or did you download the file from Food4Rhino and move it over the plugins folder manually?

**Reply by simgrouik** [reply]
I install it from the package manager

**Reply by philipbelesky** [reply]
Hi, I've just release a new version on the package manager. If you had the time, I'd appreciate it if you could try that out and see if it helps with your issue. I also found out that my build/distribution script wasn't working, so the version that was (before today) available on the package manager was also older than it should have been.

**Reply by simgrouik** [reply]
Hello Philip,
I install the latest release, and something wrong because it change nothing, I am certain that there is something with a third party plugin or driver on my computer, because my Rhino is up to date, and by just placing the componnent it does not extracts datas from the file.

![Caribou OSM3](https://user-images.githubusercontent.com/46321881/138250003-31f6d206-6086-421f-8ccb-f06809c0de65.png)
.

**Reply by philipbelesky** [reply]
Hello again. I think I've figure out at least part of the issue here (and hopefully the whole issue). I didn't realise that when C# tries to interpret a decimal number it will, unless told otherwise, expect the format to follow the user's language/culture settings. So, given you are running things in French, the error log in the Rhino command line is complaining that Caribou can't parse the latitude/longitude properly as they use periods. 

Sorry for the ignorance on my part! If you were interested in giving things another go, `v0.13.1` should fix this issue. Thanks again for your help diagnosing this.

**Comment by philipbelesky** [comment, marked as answer]
Marking this as answered as I think it should be fixed in the version I mention in my last reply. Thanks again for reporting this!
